source/Celerik.NetCore.Services/Core/ApiBuilder.cs

```diff
@@ -168,6 +168,39 @@ public ApiBuilder(IServiceCollection services, IConfiguration config)
             return this;
         }
 
+        /// <summary>
+        /// Adds a Cosmos DbContext to the current service collection
+        /// only if the the Cosmos connection string
+        /// is defined in the configuration and the service
+        /// type is ApiServiceType.ServiceEF.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">If this method
+        /// was already called.</exception>
+        internal ApiBuilder<TLoggerCategory, TDbContext> AddCosmosDb()
+        {
+            if (IsInvoked(nameof(AddCosmosDb)))
+                throw new InvalidOperationException(
+                    ServiceResources.Get("ApiBuilder.MethodAlreadyCalled", nameof(AddCosmosDb))
+                );
+
+            if (ServiceType == ApiServiceType.ServiceEF)
+            {
+                var connectionString = _config.GetConnectionString(
+                    ApiConfigKeys.CosmosDbConnectionStringName);
+
+                var cosmosConectionString = new CosmosDBConnectionString(connectionString);
+                var databaseName = _config.GetValue<string>(ApiConfigKeys.CosmosDbDatabaseName);
+
+                if (!string.IsNullOrEmpty(connectionString))
```

Comment:
put cosmosConectionString  and databaseName  declarations inside IF

Reply:
Yes! Its inside the if, line 186

Reply:
Inside the second one